crates/harp/src/vector/mod.rs

```diff
@@ -100,12 +107,20 @@ pub trait Vector {
         Rf_xlength(self.data()) as usize
     }
 
-    fn format_one(&self, x: Self::Type) -> String;
+    fn format_one(&self, x: Self::Type, options: Option<&FormatOptions>) -> String;
 
-    fn format_elt_unchecked(&self, index: isize) -> String {
+    fn format_elt_unchecked(&self, index: isize, options: Option<&FormatOptions>) -> String {
         match self.get_unchecked(index) {
-            Some(x) => self.format_one(x),
+            Some(x) => self.format_one(x, options),
             None => String::from("NA"),
         }
     }
+
+    fn format_with_options(&self, value: String, options: &FormatOptions) -> String {
```

Comment:
If this only affects the character vectors, this should be moved to the `character_vector.rs` file.